app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -937,8 +935,11 @@ class PluginApi {
    *
    **/
   addHeaderPanel(name, toggle, transformAttrs) {
+    deprecated(
+      "addHeaderPanel has been removed. Use api.addToHeaderIcons instead.",
+      { id: "discourse.add-header-panel" }
+    );
```

Comment:
Looks good, but let's comment it out until we have public documentation on Meta with migration instructions (which we can then link from the deprecation message)